src/UI/Seller/src/app/promotions/components/promotion-edit/promotion-edit.component.ts

```diff
@@ -701,13 +775,76 @@ export class PromotionEditComponent implements OnInit, OnChanges {
     }
   }
 
+  async handlePromotionAssignments(
+    promo: Promotion<PromotionXp>
+  ): Promise<void> {
+    // Assemble assignment data
+    const promoBuyers = promo.xp?.Buyers
+    const listOptions = {
+      page: 1,
+      pageSize: 100,
+      promotionID: promo.ID,
+    }
+    const currentPromotionAssignments = await this.ocPromotionService
+      .ListAssignments(listOptions)
+      .toPromise()
+    const assignmentsToDelete: string[] = []
+    const assignmentsToAdd: string[] = []
+
+    // Identify assignments to add
+    if (promoBuyers.length > 0) {
+      for (const buyerIDToAssign of promoBuyers) {
+        const matchingAssignment = currentPromotionAssignments.Items.find(
+          (assignment) => assignment?.BuyerID == buyerIDToAssign
+        )
+        if (!matchingAssignment) {
+          assignmentsToAdd.push(buyerIDToAssign)
+        }
+      }
+    }
+
+    // Identify assignments to delete
+    if (currentPromotionAssignments.Items.length > 0) {
+      for (const currentAssignment of currentPromotionAssignments.Items) {
+        const matchingAssignment = promoBuyers.find(
+          (assignment) => assignment == currentAssignment.BuyerID
+        )
+        if (!matchingAssignment) {
+          assignmentsToDelete.push(currentAssignment.BuyerID)
+        }
+      }
+    }
+    // Add assignments
+    if (assignmentsToAdd.length > 0) {
+      for (const assignment of assignmentsToAdd) {
+        await this.ocPromotionService
+          .SaveAssignment({
+            PromotionID: promo.ID,
+            BuyerID: assignment,
+          })
+          .toPromise()
+      }
+    }
+    // Delete assignments
+    if (assignmentsToDelete.length > 0) {
+      for (const assignment of assignmentsToDelete) {
+        await this.ocPromotionService
+          .DeleteAssignment(promo.ID, {
+            buyerID: assignment,
+          })
+          .toPromise()
+      }
```

Comment:
For all these assignment requests, can you push them to an array and then do `Promise.all` so that they can run in parallel? I think that would help with performance on something that may have many buyers